webapp/src/components/file_preview_modal.tsx

```diff
@@ -1,35 +1,80 @@
-import React, {FC, useCallback, useState} from 'react';
+import React, {FC, useCallback, useEffect, useState} from 'react';
+import {AnyAction} from 'redux';
+import {ThunkDispatch} from 'redux-thunk';
+import {useDispatch, useSelector} from 'react-redux';
 
 import {FileInfo} from 'mattermost-redux/types/files';
+import {GlobalState} from 'mattermost-redux/types/store';
 
-import {useDispatch, useSelector} from 'react-redux';
-
+import {updateFileEditPermission} from 'actions/file';
 import {closeFilePreview} from 'actions/preview';
-import {filePreviewModal} from 'selectors';
+import {
+    collaboraFileEditPermissionsEnabled,
+    filePreviewModal,
+    makeGetCollaboraFilePermissions,
+    makeGetIsCurrentUserFileOwner,
+} from 'selectors';
 
 import FullScreenModal from 'components/full_screen_modal';
 import WopiFilePreview from 'components/wopi_file_preview';
 import FilePreviewHeader from 'components/file_preview_header';
 
+import {FILE_EDIT_PERMISSIONS} from '../constants';
+
 type FilePreviewModalSelector = {
     visible: boolean;
     fileInfo: FileInfo;
 }
 
 const FilePreviewModal: FC = () => {
-    const dispatch = useDispatch();
+    const dispatch: ThunkDispatch<GlobalState, undefined, AnyAction> = useDispatch();
     const {visible, fileInfo}: FilePreviewModalSelector = useSelector(filePreviewModal);
     const [editable, setEditable] = useState(false);
     const toggleEditing = useCallback(() => {
         setEditable((prevState) => !prevState);
     }, [setEditable]);
 
+    const getIsCurrentUserFileOwner = makeGetIsCurrentUserFileOwner();
+    const getCollaboraFilePermissions = makeGetCollaboraFilePermissions();
+    const isCurrentUserOwner = useSelector((state: GlobalState) => getIsCurrentUserFileOwner(state, fileInfo));
+    const filePermission = useSelector((state: GlobalState) => getCollaboraFilePermissions(state, fileInfo));
+    const editPermissionsFeatureEnabled = useSelector(collaboraFileEditPermissionsEnabled);
+    const showEditPermissionChangeOption = editPermissionsFeatureEnabled && isCurrentUserOwner;
+
+    const [canChannelEdit, setCanChannelEdit] = useState(false);
+    const toggleCanChannelEdit = async () => {
+        await setCanChannelEdit((prevState: boolean) => !prevState);
```

Comment:
why this await here?

Reply:
This is required to correctly use the state in the API call below.